src/libraries/System.Security.Cryptography.Pkcs/src/Internal/Cryptography/Pal/AnyOS/ManagedPal.Decrypt.cs

```diff
@@ -146,8 +146,40 @@ private static byte[] GetAsnSequenceWithContentNoValidation(ReadOnlySpan<byte> c
                 out Exception? exception)
             {
                 exception = null;
+
+#if NET
+                try
+                {
+                    using (SymmetricAlgorithm alg = OpenAlgorithm(contentEncryptionAlgorithm))
+                    {
+                        try
+                        {
+                            alg.Key = cek;
+                        }
+                        catch (CryptographicException ce)
+                        {
+                            throw new CryptographicException(SR.Cryptography_Cms_InvalidSymmetricKey, ce);
```

Comment:
Use the same exception message that we use below if the key is an invalid size. This happens during the "RSA PKCS#1 decryption worked but produced a bogus key".